### ISSUE TITLE
Viewer doesn't display cyrillic symbols in hex, bin, dec mode

### ISSUE THREAD

**Issue by vadim-thesis** [doublecmd/doublecmd#1740]
### Version

Double Commander
Version: 1.1.16 gamma
Revision: 520
Commit: ca77fea98
Build date: 2024/06/22
Lazarus: 3.2.0.0
FPC: 3.2.2
Platform: x86_64-Linux-qt6
OS version: openSUSE Leap 15.6

Widgetset library: Qt 6.7.2, libQt6Pas 6.2.1

### Description

Viewer skips cyrillic symbols. It's not font issue. Attached screenshot shows Viewer with utf-8 file opened.

![image](https://github.com/doublecmd/doublecmd/assets/21037685/b07cf169-3a28-4725-bf86-f564d6d7d15e)

**Comment by xBZZZZ** [comment]
Are there any hex editors that display UTF-8 characters?

**Comment by alexx2000** [comment]
Not a bug. Viewer shows only single-byte encodings in these modes.

**Comment by Skif-off** [comment]
> Are there any hex editors that display UTF-8 characters?

Is this possible? Characters in UTF-8 encoding have variable length (from 1 to 6 bytes, 2 bytes for 33 alphabetic characters).

**Comment by lprot** [comment]
> Are there any hex editors that display UTF-8 characters?

Sure, wxMEdit does it:
![image](https://github.com/doublecmd/doublecmd/assets/2608545/2837b1b0-76a2-4146-ade8-ef8e88e8c8c3)

@alexx2000 can DC be extended to support this as well? Would be nice.

**Comment by vadim-thesis** [comment]
> Not a bug. Viewer shows only single-byte encodings in these modes.

My bad, taken wrong file for screenshot. Here is cp1251 file opened with explicitly set encoding:

![image](https://github.com/doublecmd/doublecmd/assets/21037685/9a23cbc4-d43f-4fef-b990-62858121b3d3)

Also please notice invalid preview frame.

**Comment by lprot** [comment]
@alexx2000 @rich2014 Indeed, DC just does not apply selected code page to the text part of the hex mode correctly
![image](https://github.com/doublecmd/doublecmd/assets/2608545/fc57e693-b7bf-4e0e-898e-b43a04013f68)
![image](https://github.com/doublecmd/doublecmd/assets/2608545/dce6d3b2-5795-4809-a9d2-b60db9195fcb)

The sample [UTF-8-demo.txt](https://github.com/user-attachments/files/16062740/UTF-8-demo.txt)

**Comment by alexx2000** [comment]
> can DC be extended to support this as well?

Maybe...

> Indeed, DC just does not apply selected code page to the text part of the hex mode correctly

Yes of course, because as I say above Viewer supports only single-byte encodings in these modes.


@vadim-thesis DC 1.2.0 has this feature. Older versions supports ASCII only.

> Also please notice invalid preview frame.

Preview is an independent entity, it always use system encoding. It is a same thing as thumbnail view in the file panels.
